src/Orleans/AssemblyLoader/AssemblyLoader.cs

```diff
@@ -64,19 +64,20 @@ internal class AssemblyLoader
                     pathNameCriteria,
                     reflectionCriteria,
                     logger);
-
-            int count = 0;
+            
+            var loadedAssemblies = new List<Assembly>();
             List<string> discoveredAssemblyLocations = loader.DiscoverAssemblies();
             foreach (var pathName in discoveredAssemblyLocations)
             {
                 loader.logger.Info("Loading assembly {0}...", pathName);
+
                 // It is okay to use LoadFrom here because we are loading application assemblies deployed to the specific directory.
                 // Such application assemblies should not be deployed somewhere else, e.g. GAC, so this is safe.
-                Assembly.LoadFrom(pathName);
-                ++count;
+                loadedAssemblies.Add(Assembly.LoadFrom(pathName));
```

Comment:
We should put it around try/catch to not to fail the whole loading loop if something fails to load.